Classes/Task/ReIndexTaskAdditionalFieldProvider.php

```diff
@@ -111,25 +111,28 @@ public function getAdditionalFields(
         $task,
         SchedulerModuleController $schedulerModule
     ) {
-        $this->isTaskInstanceofReIndexTask($task);
-
-        $this->initialize($taskInfo, $task, $schedulerModule);
         $additionalFields = [];
 
-        $additionalFields['site'] = [
-            'code' => Site::getAvailableSitesSelector('tx_scheduler[site]',
-                $this->site),
-            'label' => 'LLL:EXT:solr/Resources/Private/Language/locallang.xlf:field_site',
-            'cshKey' => '',
-            'cshLabel' => ''
-        ];
-
-        $additionalFields['indexingConfigurations'] = [
-            'code' => $this->getIndexingConfigurationSelector(),
-            'label' => 'Index Queue configurations to re-index',
-            'cshKey' => '',
-            'cshLabel' => ''
-        ];
+        if ($this->isTaskInstanceofReIndexTask($task)) {
```

Comment:
I would propose to keep the usages of isTaskInstanceofIndexQueueWorkerTask as it was because otherwise we just introduce another level maybe it is an option to rename isTaskInstanceofIndexQueueWorkerTask in checkIsTaskInstanceofIndexQueueWorkerTask ?